shapely/strtree.py

```diff
@@ -75,44 +72,26 @@ class STRtree:
     None
     """
 
-    def __init__(self, geoms):
-        warn(
-            "STRtree will be completely changed in 2.0.0. The exact API is not yet decided, but will be documented before 1.8.0",
-            ShapelyDeprecationWarning,
-            stacklevel=2,
-        )
-        # filter empty geometries out of the input
-        geoms = [geom for geom in geoms if not geom.is_empty]
-        self._n_geoms = len(geoms)
-
-        self._init_tree_handle(geoms)
-
-        # Keep references to geoms.
-        self._geoms = list(geoms)
-
-    def _init_tree_handle(self, geoms):
-        node_capacity = 10
-        self._tree_handle = lgeos.GEOSSTRtree_create(node_capacity)
-        for geom in geoms:
-            lgeos.GEOSSTRtree_insert(self._tree_handle, geom._geom, ctypes.py_object(geom))
-
-    def __getstate__(self):
-        state = self.__dict__.copy()
-        del state["_tree_handle"]
-        return state
-
-    def __setstate__(self, state):
-        self.__dict__.update(state)
-        self._init_tree_handle(self._geoms)
-
-    def __del__(self):
-        if self._tree_handle is not None:
-            try:
-                lgeos.GEOSSTRtree_destroy(self._tree_handle)
-            except AttributeError:
-                pass  # lgeos might be empty on shutdown.
-
-            self._tree_handle = None
+    def __init__(self, geoms, items=None, leafsize=10):
+        self.geometries = np.asarray(geoms, dtype=np.object_)
+        self._tree = lib.STRtree(self.geometries, leafsize)
+
+        has_custom_items = items is not None
+        if has_custom_items:
+            items = np.asarray(items)
+        else:
+            items = np.arange(len(self.geometries))
+        self._items = items
+        self._has_custom_items = has_custom_items
+
+    def __len__(self):
+        return self._tree.count
+
+    def __reduce__(self):
+        if self._has_custom_items:
+            return (STRtree, (self.geometries, self._items))
+        else:
+            return (STRtree, (self.geometries, ))
 
     def query(self, geom):
```

Comment:
This function now returns an `np.array` instead of a list which is causing the doctest to fail.

Reply:
docstring currently has "array or list of geometry objects.". Perhaps it should be updated to say just "array".

Reply:
Ah, yes, updated the docstrings.